### PR TITLE
Seller services controller

### PR DESCRIPTION
**Pull request by tavisca-acharpe**

### CODE REVIEW

File: src/OnlineRetailPortal.Web/Models/AddProductRequest.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OnlineRetailPortal.Web
+{
+    public class AddProductRequest
+    {
+        public string UserId { get; set; }
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Models/AddProductRequest.cs]: How will you differentiate between user id passed in header and user id taken in request.
Shouldn’t the field name be postedBy or SellerId?
Just a question

File: src/OnlineRetailPortal.Web/Models/Category.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OnlineRetailPortal.Web
+{
+    public enum Category
+    {
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Models/Category.cs:8]: What should be the default category?
Should it be properties or something generic?

File: src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            .Length(2, 20)
+            .WithMessage("The Title cannot be less than 2 characters and more than  20 characters.");
+
+            RuleFor(x => x.Category)
+            .NotEmpty()
+            .WithMessage("The Categorye cannot be blank.");
+
+            RuleFor(x => x.Price.Amount)
+            .NotEmpty()
+            .InclusiveBetween(1, 10000000)
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs]: What if this is the property for sale?
We should not restrict outer bound 
Or we need to expose it as a part of contract that products beyond 10000000
Let’s discuss this tomorrow

File: src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace OnlineRetailPortal.Web.Validations
+{
+    public class AddProductRequestValidator : AbstractValidator<AddProductRequest>
+    {
+        public AddProductRequestValidator()
+        {
+            RuleFor(x => x.UserId)
+            .NotEmpty()
+            .WithMessage("The User ID  cannot be blank.");
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs]: What if the userId field is null?
You should also check for null
Also we need to discuss the format of how are we collecting all these messages and returning as a part if error response

File: src/OnlineRetailPortal.Web/Controllers/ProductsController.cs [hunk 5]
```diff
@@ -18,13 +22,24 @@ public ProductsController(IProductService productService=null)
...
             //return new GetProductResponse();
         }
+
+        [HttpPost("Add")]
+        public async Task<AddProductResponse> AddProductAsync([FromBody] AddProductRequest request)
+        {
+            AddProductRequestValidator validator = new AddProductRequestValidator();
+            validator.Validate(request);
+            var response = await _productService.AddProductAsync(request.ToDataContract());
+            return response.ToUserContract();
```
Comment (Tavisca-smohite) [inline src/OnlineRetailPortal.Web/Controllers/ProductsController.cs]: Why ToUserContract()??

**Review by Tavisca-smohite — CHANGES_REQUESTED** [review]
Please add test cases for your impl

File: src/OnlineRetailPortal.Web/Controllers/ProductsController.cs (hunk 5 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Controllers/ProductsController.cs]: ToUser()

File: src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs (hunk 3 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs]: ok

File: src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs (hunk 4 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs]: ok

File: src/OnlineRetailPortal.Web/Models/Category.cs (hunk 2 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Models/Category.cs:8]: it is mandatory field

File: src/OnlineRetailPortal.Web/Models/AddProductRequest.cs (hunk 1 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Models/AddProductRequest.cs]: done

File: src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs (hunk 3 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs]: done

File: src/OnlineRetailPortal.Web/Controllers/ProductsController.cs (hunk 5 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Web/Controllers/ProductsController.cs]: done

**Pull request closed without merging**